contents/blog/version-30-upgrades/index.mdx

```diff
@@ -0,0 +1,82 @@
+---
+title: Marquez 0.30.0 offers a big performance bump, with some assembly required
+date: 2023-02-03
+author: Michael Robinson
+image: ./image.svg
+banner: ./banner.svg
+description: How to ensure a smooth transition to the latest version of Marquez. 
+---
+
+## Marquez 0.30.0
+
+[Marquez 0.30.0](https://github.com/MarquezProject/marquez/releases/tag/0.30.0), 
+which we released on 2023-01-31, contains some exciting new features, including 
+[column-level lineage](https://github.com/MarquezProject/marquez/pull/2293) and 
+a [soft delete option](https://github.com/MarquezProject/marquez/pull/2343) in 
+the UI.
+
+Along with these long-awaited UI upgrades come some significant performance 
+improvements to the API. In prior versions of Marquez, accessing OpenLineage 
+facets required direct queries against the OpenLineage events table, which could 
+be quite costly depending on the size of this table. Some users have events 
+tables with more than 100K rows -- millions of rows, in fact. To address this 
+performance issue, Marquez 30.0 has new tables for dataset, job and run facets.
+
+So far, so simple, right? Protecting against data loss and ensuring a smooth 
+experience for users was not so simple, in fact. We needed a way to migrate the 
+data while not only avoiding loss but also keeping downtime to a minimum. The 
+solution that [Willy](https://github.com/wslulciuc) and 
+[Pawel](https://github.com/pawel-big-lebowski) found (with input from 
+[Mike Collado](https://github.com/collado-mike)) was to keep the data in the 
+events table until the migration process is complete and to create views on top 
+of the table for accessing the facets. After migration is complete, the views 
+are replaced to point to the new tables. Using views provides safeguards and 
+minimizes downtime during the migration. 
+
+## Migration Procedure
+
+As the 
+[migration plan](https://github.com/MarquezProject/marquez/blob/main/api/src/main/resources/marquez/db/migration/V57__readme.md) 
+says, users whose `lineage_events` tables have up to 100K rows can set it and 
+forget it -- no extra work is required beyond upgrading and restarting 
+Marquez, but be prepared for a brief downtime while a standard Flyway migration 
+fills the newly created tables automatically.
+
+For heavy users whose `lineage_events` tables have more than 100K rows, a 
+different process will be used. Instead of a standard migration, the process will 
+happen asynchronously so that the user can start the API and consume new 
+OpenLineage events as usual. Note that some API calls might return irrelevant 
+results during the migration, especially if the output is based on facets.
```

Comment:
In fact this will not happen. Until asynchronous migration has finished views based on existing old tables are used. So there is no downtime on write nor read side.

It's just that performance boost will be present after the migration is finished.